src/System.DirectoryServices/tests/System/DirectoryServices/ActiveDirectory/ActiveDirectoryInterSiteTransportTests.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.ComponentModel;
+using Xunit;
+
+namespace System.DirectoryServices.ActiveDirectory.Tests
+{
+    public class ActiveDirectoryInterSiteTransportTests
+    {
+        [Fact]
+        public void FindByTransportType_NullContext_ThrowsArgumentNullException()
+        {
+            AssertExtensions.Throws<ArgumentNullException>("context", () => ActiveDirectoryInterSiteTransport.FindByTransportType(null, ActiveDirectoryTransportType.Rpc));
+        }
+
+        [Fact]
+        public void FindByTransportType_ForestNoDomainAssociatedWithoutName_ThrowsActiveDirectoryOperationException()
+        {
+            var context = new DirectoryContext(DirectoryContextType.Forest);
+            Assert.Throws<ActiveDirectoryOperationException>(() => ActiveDirectoryInterSiteTransport.FindByTransportType(context, ActiveDirectoryTransportType.Rpc));
+        }
+
+        [Theory]
+        [InlineData("\0")]
+        [InlineData("server:port")]
+        public void FindByTransportType_ForestNoDomainAssociatedWithName_ThrowsActiveDirectoryOperationException(string name)
+        {
+            var context = new DirectoryContext(DirectoryContextType.Forest, name);
+            Assert.Throws<ArgumentException>("context", () => ActiveDirectoryInterSiteTransport.FindByTransportType(context, ActiveDirectoryTransportType.Rpc));
```

Comment:
AssertExtensions.Throws .. can you please check for others?
We should just clone our own Assert class and fix it..